moduleframework/helpers/openshift_helper.py

```diff
@@ -0,0 +1,268 @@
+                                                                                   oc_user,
+                                                                                   oc_passwd),
+                                 verbose=common.is_not_silent())
+        common.print_debug(oc_output.stdout)
+        return oc_output.exit_status
+
+    def tearDown(self):
+        """
+        Cleanup environment and call also cleanup from config
+
+        :return: None
+        """
+        super(OpenShiftHelper, self).tearDown()
+        if common.get_if_do_cleanup():
+            # TODO will be implemented later on. I have to find a usecase what to remove
+            pass
+
+    def _get_ip_instance(self):
+        """
+        It gets and IP address of an application from of OpenShift POD.
+        :return: True: getting IP address was successful
+                 False: getting IP address was not successful
+        """
+        oc_get_service = self.runHost("oc get service -o json")
+        service = self._convert_string_to_json(oc_get_service.stdout)
+        try:
+            for svc in service["items"]:
+                if "clusterIP" in svc.get("spec"):
+                    common.trans_dict['GUESTIPADDR'] = svc.get("spec").get("clusterIP")
+            return True
+        except KeyError as e:
+            common.print_info(e.message)
+            return False
+        except IndexError as e:
+            common.print_info(e.message)
+            return False
+
+    def start(self):
+        """
+        starts the OpenShift application
+
+        :param args: Do not use it directly (It is defined in config.yaml)
+        :param command: Do not use it directly (It is defined in config.yaml)
+        :return: None
+        """
+        if not self._app_exists():
+            self._create_app()
+            # Verify application is really deploy and prepared for testing.
+            self._verify_pod()
+            # TODO fix in case IP is not present. We should failed.
+            self._get_ip_instance()
+
+    def stop(self):
+        """
+        Stops the OpenShift
+
+        :return: None
+        """
+        if self.status():
+            try:
+                self._app_remove()
+            except Exception as e:
+                common.print_info(e, "OpenShift application already removed")
+                pass
+
+    def status(self):
+        """
+        get status of an OpenShift
+
+        :return: bool
+        """
+        if self._app_exists():
+            return True
+        else:
+            return False
+
+    def run(self, command="ls /", **kwargs):
+        """
+        Run command inside module, all params what allows avocado are passed inside shell,ignore_status, etc.
+
+        :param command: str
+        :param kwargs: dict
+        :return: avocado.process.run
+        """
+        return self.runHost("%s" % common.sanitize_cmd(command), **kwargs)
```

Comment:
Are you sure that this does what you expect?
this run command on host.
`run` method of helper should do command on __GUEST__ in this case on openshift instance in case of docker there is for example `docker exec INSTANCE_ID command` so in case of openshift I expect something similar like `oc exec ...` 

it is connected with more missing special methods for openshift, like `CopyFrom` and `CopyTo` now are inherited from CommonFunctions, what leads to `host to host` implementation, I expect there also some special `oc ` command, or do it via some ssh, or whatever how it is connected together

Probably my mistake were that I've defined these functions leads to `localhost` operations, instead of raising not defined exceptions, and implement these methods where it make sense like in `RpmHelper, ContainerHelper` etc...